src/Api/Middleware/ResolveRouteFromName.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Api\Middleware;
+
+use Flarum\Http\RouteCollection;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ResolveRouteFromName implements MiddlewareInterface
+{
+    public function __construct(RouteCollection $routeCollection)
+    {
+        $this->routeCollection = $routeCollection;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $route = $this->routeCollection->getRoutes()[$request->getAttribute('routeName')];
+
+        $request = $request->withMethod($route['method']);
```

Comment:
How will this work with routes that might be available via multiple methods? Can we predict which one will be chosen? Should there be a way to specify?

I wouldn't mind having `$client->post()`, `$client->get()` and such and pass the method as an additional request attribute.

Reply:
Converted to use paths, as per our chat discussion